### PR TITLE
initial commit

### PR DESCRIPTION
**Pull request by christianalfoni**
Okay, so I have refactored the operators to use the [cerebral-scheme-parser](https://github.com/cerebral/cerebral-scheme-parser) project.

I have removed features that was not documented. Things like async grabbing things, array use on `when` etc. Since it was not documented it is unlikely to be in use. And if they are it is possible to create a normal action or use the `cerebral-scheme-parser` directly. I just want to limit the API as much as possible, and we can rather increase "on request". It has been really cool to see all the possibilities though :)

I have also added tests for ensuring that the different operators can use inline schemes etc. Also throwing errors on wrong usage is pretty good now.

I also changed the displayNames of the actions to avoid cluttering the debugger. Some of them still has extra info, but most of them explains very well what they do when you look at the state mutation or output in the debugger. Again, just want to limit and rather add things in "on request". 

But the core thing here is to provide the `schemeParser` directly through Cerebral:

``` js
import {parseScheme} from 'cerebral'
```

The current docs on the repo ([cerebral-scheme-parser](https://github.com/cerebral/cerebral-scheme-parser)) will be moved to `operators` page of website. This will allow developers to do whatever they want and the API is a bit simpler and not as opinionated  as the previous compiler. Think we will see more contributions and ideas :)

### CODE REVIEW

**Comment by christianalfoni** [comment]
Do not merge this until we have released 1.0 of Cerebral

File: src/set.js [hunk 1]
```diff
@@ -1,17 +1,20 @@
...
-  const setValue = setCompiler(path)
+  const pathScheme = parseScheme(path)
 
-  const set = function set (args) {
-    const response = setValue(args, value)
-    if (response && typeof response.then === 'function') {
-      response.then(args.output.success).catch(args.output.error)
-    }
+  if (pathScheme.target !== 'state') {
+    throw new Error('Cerebral operator SET - The path: "' + path + '" does not target "state"')
```
Comment (garth) [inline src/set.js]: @christianalfoni Is there a reason why this can't work on output?

File: package.json [hunk 2]
```diff
@@ -41,6 +41,7 @@
     "watch": "^0.19.1"
   },
   "dependencies": {
+    "cerebral-scheme-parser": "^0.1.1",
     "cerebral-url-scheme-compiler": "^0.5.2",
```
Comment (fmal) [inline package.json]: I guess `cerebral-url-scheme-compiler` could be removed from the deps :)

File: src/set.js (hunk 1 above)
Comment (christianalfoni) [inline src/set.js]: Ah, no, not really :) I just could not imagine a use case for it?

Comment (garth) [inline src/set.js]: I've used it in the past when I want to setup input data for an action.

**Comment by garth** [comment]
Bearing in mind that this change will probably break cerebral-addons, do we wish to continue with addons?

File: src/set.js (hunk 1 above)
Comment (christianalfoni) [inline src/set.js]: Okay, cool, no reason to not have it :) Let me make an update! Thanks for reviewing it!

Comment (christianalfoni) [inline src/set.js]: Added now :)

**Comment by christianalfoni** [comment]
@garth Hm, does it break addons? Is it not self contained, depending on scheme-compiler etc.? Or does it use anything from `cerebral/operators`?

**Comment by christianalfoni** [comment]
Okay, also removed lodash and compiler deps here now

**Comment by christianalfoni** [comment]
I checked the `addons` package and can not find any references to `cerebral/operators` and it has its own dependency to compiler so I am very sure this will be okay :)

**Comment by Guria** [comment]
@christianalfoni you shouldn't remove lodash because it is already fixed in master. please rebase a pr

**Comment by christianalfoni** [comment]
Okay, fixed :)

**Comment by garth** [comment]
Addons relies on the getter/setter API, which would be changed with this
upset. So we need to decide what to do with addons.

**Comment by christianalfoni** [comment]
@garth But addons still has access to getter/setter API through the compiler dependency? Sorry, do not see how addons/operators has anything to do with each other?

File: package.json [hunk 3]
```diff
@@ -41,6 +41,7 @@
     "watch": "^0.19.1"
   },
   "dependencies": {
-    "cerebral-url-scheme-compiler": "^0.5.2"
+    "cerebral-scheme-parser": "^0.1.1",
+    "lodash": "^4.14.2"
```
Comment (Guria) [inline package.json]: :) remove it please

**Comment by christianalfoni** [comment]
Haha, okay, NOW :)

**Comment by garth** [comment]
Addons has a bunch of functions that will no longer work. But that's ok we just need to deprecate them now before this is released, or maybe deprecate the entire lib.

**Comment by christianalfoni** [comment]
@garth Haha, I still do not understand how addons can be affected by this? :) If you bring in addons and use addons stuff that will just run inside addons. If you use `cerebral/operators` that will only run stuff inside operators... or is current operators using addons?

**Comment by Guria** [comment]
I see only import of `copy` in tests of addons

**Comment by Guria** [comment]
hmm
@christianalfoni I just checked out tests with this (yet unreleased) version of operators.
and it fails on tests:

```
/home/ubuntu/workspace/cerebral-addons/node_modules/cerebral-operators/node_modules/cerebral-scheme-parser/index.js:2
  var targetMatch = path.match(/(^.*?):/);
```

does it mean that you are breaking operators api?
because tests of addons just use it as any other consumer. it shouldn't fail

**Comment by Guria** [comment]
I have merged it, but haven't released yet. So lets investigate compatibility issues and safe upgrade path.

**Comment by Guria** [comment]
@christianalfoni @garth can we check now if some of addons were passed to operator and throw a good error there?

**Comment by garth** [comment]
Yes, I think that is must be done by operators. I could add a property to the addon getters/setters that indicates where it came from.

**Comment by Guria** [comment]
hmm, what if user will update only cerebral and not addons? couldn't we detect somehow using duck-typing?

**Pull request merged**